wormchain/x/wormhole/keeper/query_plugins.go

```diff
@@ -0,0 +1,78 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+func NewCustomQueryHandler(keeper Keeper) *wasmkeeper.QueryPlugins {
+	return &wasmkeeper.QueryPlugins{
+		Custom: WormholeQuerier(keeper),
+	}
+}
+
+type WormholeQuery struct {
+	VerifyQuorum    *verifyQuorumParams    `json:"verify_quorum,omitempty"`
+	VerifySignature *verifySignatureParams `json:"verify_signature,omitempty"`
+	CalculateQuorum *calculateQuorumParams `json:"calculate_quorum,omitempty"`
+}
+
+type verifyQuorumParams struct {
+	Data             []byte           `json:"data"`
+	GuardianSetIndex uint32           `json:"guardian_set_index"`
+	Signatures       []*vaa.Signature `json:"signatures"`
+}
+
+type verifySignatureParams struct {
+	Data             []byte         `json:"data"`
+	GuardianSetIndex uint32         `json:"guardian_set_index"`
+	Signature        *vaa.Signature `json:"signature"`
+}
+
+type calculateQuorumParams struct {
+	GuardianSetIndex uint32 `json:"guardian_set_index"`
+}
+
+func WormholeQuerier(keeper Keeper) func(ctx sdk.Context, data json.RawMessage) ([]byte, error) {
+	return func(ctx sdk.Context, data json.RawMessage) ([]byte, error) {
+		var wormholeQuery WormholeQuery
+		err := json.Unmarshal(data, &wormholeQuery)
+		if err != nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
+		}
+
+		if wormholeQuery.VerifyQuorum != nil {
+			// handle the verify quorum query
+			err := keeper.VerifyQuorum(ctx, wormholeQuery.VerifyQuorum.Data, wormholeQuery.VerifyQuorum.GuardianSetIndex, wormholeQuery.VerifyQuorum.Signatures)
+			if err != nil {
+				return nil, err
+			}
+			return []byte{}, nil
+		}
+		if wormholeQuery.VerifySignature != nil {
+			// handle the verify signature query
+			err := keeper.VerifySignature(ctx, wormholeQuery.VerifySignature.Data, wormholeQuery.VerifySignature.GuardianSetIndex, wormholeQuery.VerifySignature.Signature)
+			if err != nil {
+				return nil, err
+			}
+			return []byte{}, nil
```

Comment:
There is no marshaled type to return?

Reply:
In the proposed API https://github.com/wormhole-foundation/wormhole/issues/1881 the return types for these queries are `cosmwasm_std::Empty` -- https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Empty.html.

Not sure exactly what they serialize to but empty byte array seems to be a reasonable placeholder until https://github.com/wormhole-foundation/wormhole/pull/1920 is merged and I can test the contract e2e.